src/hidpp20.h

```diff
@@ -83,6 +83,7 @@ hidpp20_device_destroy(struct hidpp20_device *device);
 
 enum {
```

Comment:
can you make this a named enum place (e.g. `enum hiddp20_quirks`), that way you can use it as argument type.